src/tests/harness/BenchmarkDotNet.Extensions.Tests/UniqueValuesGeneratorTests.cs

```diff
@@ -72,5 +72,115 @@ public void GetNonDefaultValueReturnsNonDefaultValue()
             Assert.NotEqual(default(int), ValuesGenerator.GetNonDefaultValue<int>());
             Assert.NotEqual(default(string), ValuesGenerator.GetNonDefaultValue<string>());
         }
+
```

Comment:
Add all the missing tests that assert that we'll be able to generate values using the exposed public methods. For each type (other than `bool`) we will call `Supports<T>();` of the correct type which internally will do the following to assert behavior of the public methods. For `bool` there's only two unique values, so we only assert the array construction, not the ones that test distinct or non-default values.
```
SupportsArray<T>();
SupportsNonDefaultValue<T>();
SupportsArrayOfUniques<T>();
```

Reply:
Extended the tests to include bool and also added `Dictionary<TKey, TValue>` tests for all types (with `TValue` being the same type as `TKey` and ALSO `string`)